domain/src/main/kotlin/com/krzykrucz/elesson/currentlesson/domain/topic/domain/TopicPersistence.kt

```diff
@@ -0,0 +1,6 @@
+package com.krzykrucz.elesson.currentlesson.domain.topic.domain
+
+import arrow.fx.IO
+
+typealias CountFinishLessons = () -> IO<FinishedLessonsCount>
+typealias PersistInProgressLesson = (InProgressLesson) -> IO<Unit>
```

Comment:
I just wonder if `IO` and similar concepts should be part of the domain at all. One the one hand it's useful to express the purpose explicitly in this case `IO` effect as the operation on some external environment but on the other hand, it doesn't look like a part of a domain language :man_shrugging:  towards for directions are we going to push the model? Personally, I prefer to have domain clean so that usage of `Arrow` does not leak into the domain

Reply:
I think that the purpose of modelling it in such way is to encode as much information in types as possible. Without the IO, we do not have any context and we can assume that all happens instantly.   I think it is somehow a part of domain, it tell us that that function is talking to external system.

Reply:
we choose to keep IO as a part of domain, as Wlaschin does

Reply:
I'm partially convinced here but still need to read that part of book :wink: